File: Wms.Api/Dto/StockOut/StockOutCreateUpdate/StockOutCreateUpdateDto.cs
using Wms.Api.Dto.StockOut.StockOutCreateUpdate;
using Wms.Api.Model;

namespace Wms.Api.Dto.StockOut.StockOutCreateUpdateDto
{
    public class StockOutCreateUpdateDto 
    { 
        public required string Number { get; set; }
        public required string DONumber { get; set; }
        public required string SONumber { get; set; }
        public required string ToLocation { get; set; }
        public StockOutStatusEnum Status { get; set; }
        public Guid WarehouseId { get; set; }
        public ICollection<StockOutItemCreateUpdateDto>? StockOutItems { get; set; }
    }
}

File: Frontend/src/interfaces/v12/stockout/stockOutCreateUpdate/stockOutCreateUpdateDto.ts
import { guid } from "types/guid";

export interface StockOutCreateUpdateDto {
  number?: string | null;
  doNumber: string;
  soNumber: string;
  warehouseId: guid;
  toLocation: string;
  stockOutItems: StockOutItemCreateUpdate[];
}
export interface StockOutItemCreateUpdate {
  productId: guid;
  locationId: guid;
  quantity: number;
}
